kornia/losses/focal.py

```diff
@@ -207,9 +211,17 @@ def binary_focal_loss_with_logits(
     if input.size(0) != target.size(0):
         raise ValueError(f'Expected input batch_size ({input.size(0)}) to match target batch_size ({target.size(0)}).')
 
+    if pos_weight is None:
+        pos_weight = torch.ones(input.size(-1), device=input.device)
```

Comment:
```suggestion
        pos_weight = torch.ones(input.size(-1), device=input.device, dtype=input.dtype)
```
we should infer also dtypes